### ISSUE TITLE
crawler print warning with observation and reward nan

### ISSUE THREAD

**Issue by chagmgang** [Unity-Technologies/ml-agents#1832]
```
WARNING:mlagents.envs:An agent had a NaN reward for brain CrawlerStaticLearning
WARNING:mlagents.envs:An agent had a NaN observation for brain CrawlerStaticLearning
```
An error occurs on the training.
This error is printed with tensorflow training.
Tensorflow-gpu 1.12.0 version, python 3.6 is used.

**Comment by ervteng** [comment]
Hi @chagmgang, this is a known bug with the Crawler environment. We believe there is a divide-by-zero somewhere in the game code. 

Did it stop learning, or did your crawler learn how to walk anyways?

**Comment by chagmgang** [comment]
The crawler environment is not working well. So in order to prove my algorithm, I did my ppo2 algorithm to walker. The result with tensorboard is here.

![image](https://user-images.githubusercontent.com/37325825/54578601-a7ccec80-4a43-11e9-839d-a4fc91e057af.png)

My tensorflow repository including my algorithms is [https://github.com/RLOpensource/tensorflow_RL](https://github.com/RLOpensource/tensorflow_RL)

**Comment by ervteng** [comment]
Hi @chagmgang, I see. You're using your own trainer code w/ ml-agents, and it looks like at least it is training with walker. Was your walker able to walk?

If you're interested in a bunch of similar environments with different walker/crawler configurations, check out the Marathon Envs: https://github.com/Unity-Technologies/marathon-envs

**Comment by chagmgang** [comment]
My walker may be walk well because same algorithm to bipedalwalker makes good walk mechanism.
So Thanks for answering my question about environment.

**Comment by ervteng** [comment]
Hey @chagmgang, I'm trying to replicate your issue with Crawler. How did you manage to get NaNs out? Was it using the ML-Agents trainer, or your trainer?

Thanks!

**Comment by ervteng** [comment]
Hi @chagmgang, we believe it is an issue with Unity's Vector3.normalize function in the C# code. On our build of Unity (2017.4) no NaNs are outputted. Which version of Unity are you using?

Thanks

**Comment by ervteng** [comment]
This issue should be fixed in the new Crawler environment in v0.10. Closing this bug.
